src/FakeItEasy/Expressions/ParsedCallExpression.cs

```diff
@@ -34,5 +34,28 @@ internal class ParsedCallExpression
         public IEnumerable<ParsedArgumentExpression> ArgumentsExpressions { get; }
 
         public object CallTarget => this.callTarget.Value;
+
+        // Expression evaluation optimized for a very common case: 
+        // - it's a 'MemberAccess' expression on a local lambda closure (A.CallTo(() => fakeObj.doSomething(args));

```

Comment:
@thomaslevesque, I did not know what I was talking about. I think I figured it out and sent #852 with (I think) more explicit and still correct comments…